File: alu/scalat/src/main/scala/ALUWrapper.scala
import com.ut.UT_ALU

import java.math.BigInteger;

class ALUWrapper {
    private val alu = new UT_ALU()

    def process(a: Int, b: Int, sel: Int): Int = {
      alu.a.Set(a)
      alu.b.Set(b)
      alu.alu_sel.Set(sel)
      alu.Step()
      alu.alu_out.Get().intValue()
    }


}

File: alu/tiramisu/src/main/java/org/example/ALUWrapper.java
package org.example;

import com.ut.UT_ALU;
import org.xaspect.AutoDUTDao;

public class ALUWrapper {

    @AutoDUTDao
    ALUDutDao dao;

    UT_ALU alu;

    ALUWrapper() {
        alu = new UT_ALU();
        dao.bind(alu);
    }

    public int process(int a, int b, int sel) {
        dao.postIn(a, b, sel);
        alu.Step();
        return dao.getOut();
    }

}
